### PR TITLE
feat: [CDS-44054]: Improve Azure WebApp settings update logic

### PR DESCRIPTION
**Pull request by abosii-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 11 Issues 
* Code issue 9 Code issues 
* Anti-pattern 2 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38357)

File: 125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppSlotDeploymentStep.java [hunk 1]
```diff
@@ -212,7 +217,7 @@ public StepResponse finalizeExecutionWithSecurityContext(Ambiance ambiance, Step
     return stepResponseBuilder.stepOutcome(stepOutcome).build();
   }
 
-  private void updateStageExecutionDetails(
+  private void updatStageExecutionDetails(
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppSlotDeploymentStep.java:220]: minor typo here

**Review by vlica-harness — APPROVED** [review]
> Reviewed 15 of 16 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)* status: :shipit: complete! ✓

**Review by tmacari-harness — CHANGES_REQUESTED** [review]
> Reviewed 16 of 16 files at r1, 1 of 1 files at r2, 14 of 14 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppStepHelper.java` line 509 at r3](https://reviewable.io/reviews/harness/harness-core/38357#-NE6gCpv0KyFDMW4ZrVh:-NE6gCpv0KyFDMW4ZrVi:bnq3s3i) ([raw file](https://github.com/harness/harness-core/blob/9a93047e83fceae6e0e392bf007c3b5d1c601964/125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppStepHelper.java#L509)):*
> ```Java
>       AzureWebAppsStageExecutionDetails executionDetails =
>           (AzureWebAppsStageExecutionDetails) stageExecutionInfoList.get(0).getExecutionDetails();
>       log.info("Last successful deployment's artifact found with pipeline executionId: {}",
> ```

lets change the logging once we use it not only for artifact config

**Review by vlica-harness — DISMISSED** [review]

**Review by abosii-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppStepHelper.java` line 509 at r3](https://reviewable.io/reviews/harness/harness-core/38357#-NE6gCpv0KyFDMW4ZrVh:-NEZsYIv7is-Dkn_8SSR:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/9a93047e83fceae6e0e392bf007c3b5d1c601964/125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppStepHelper.java#L509)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

lets change the logging once we use it not only for artifact config
</blockquote></details>

Done

**Review by tmacari-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)* status: :shipit: complete! ✓

**Review by vlica-harness — DISMISSED** [review]
> Reviewed 14 of 14 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)* status: :shipit: complete! ✓

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357)* status: :shipit: complete! ✓

**Comment by abosii-harness** [comment]
trigger messagecheck

**Review by vlica-harness — APPROVED** [review]

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 6 of 16 files at r1, 14 of 14 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38357#-:-NEjYKPPCqXL2xHJF_aH:bnfp4nl)* status: :shipit: complete! ✓

**Comment by abosii-harness** [comment]
trigger messagecheck, codebasehash

**Pull request merged**